asv_bench/benchmarks/pandas/frame_methods.py

```diff
@@ -0,0 +1,286 @@
+        self.df2 = DataFrame(
+            {
+                c: {
+                    0: np.random.randint(0, 2, N).astype(np.bool_),
+                    1: np.random.randint(0, N, N).astype(np.int16),
+                    2: np.random.randint(0, N, N).astype(np.int32),
+                    3: np.random.randint(0, N, N).astype(np.int64),
+                }[np.random.randint(0, 4)]
+                for c in range(N)
+            }
+        )
+
+    def time_rename_single(self):
+        self.df.rename({0: 0})
+
+    def time_rename_axis0(self):
+        self.df.rename(self.dict_idx)
+
+    def time_rename_axis1(self):
+        self.df.rename(columns=self.dict_idx)
+
+    def time_rename_both_axes(self):
+        self.df.rename(index=self.dict_idx, columns=self.dict_idx)
+
+    def time_dict_rename_both_axes(self):
+        self.df.rename(index=self.dict_idx, columns=self.dict_idx)
+
+
+class Repr:
+    def setup(self):
+        nrows = 10000
+        data = np.random.randn(nrows, 10)
+        arrays = np.tile(np.random.randn(3, nrows // 100), 100)
+        idx = MultiIndex.from_arrays(arrays)
+        self.df3 = DataFrame(data, index=idx)
+        self.df4 = DataFrame(data, index=np.random.randn(nrows))
+        self.df_tall = DataFrame(np.random.randn(nrows, 10))
+        self.df_wide = DataFrame(np.random.randn(10, nrows))
+
+    def time_repr_tall(self):
+        repr(self.df_tall)
+
+    def time_frame_repr_wide(self):
+        repr(self.df_wide)
+
+
+class MaskBool:
+    def setup(self):
+        data = np.random.randn(1000, 500)
+        df = DataFrame(data)
+        df = df.where(df > 0)
+        self.bools = df > 0
+        self.mask = isnull(df)
+
+    def time_frame_mask_bools(self):
+        self.bools.mask(self.mask)
+
+    def time_frame_mask_floats(self):
+        self.bools.astype(float).mask(self.mask)
+
+
+class Isnull:
+    def setup(self):
+        N = 10**3
+        self.df_no_null = DataFrame(np.random.randn(N, N))
+
+        sample = np.array([np.nan, 1.0])
+        data = np.random.choice(sample, (N, N))
+        self.df = DataFrame(data)
+
+        sample = np.array(list(string.ascii_letters + string.whitespace))
+        data = np.random.choice(sample, (N, N))
+        self.df_strings = DataFrame(data)
+
+        sample = np.array(
+            [
+                NaT,
+                np.nan,
+                None,
+                np.datetime64("NaT"),
+                np.timedelta64("NaT"),
+                0,
+                1,
+                2.0,
+                "",
+                "abcd",
+            ]
+        )
+        data = np.random.choice(sample, (N, N))
+        self.df_obj = DataFrame(data)
+
+    def time_isnull_floats_no_null(self):
+        isnull(self.df_no_null)
+
+    def time_isnull(self):
+        isnull(self.df)
+
+    def time_isnull_strngs(self):
+        isnull(self.df_strings)
+
+    def time_isnull_obj(self):
+        isnull(self.df_obj)
+
+
+class Dropna:
```

Comment:
How much does this differ from fillna which we already test?

Reply:
`dropna` always goes through axis-wise mapping whereas `fillna` have multiple options to get into (axis-wise mapping, binary op, block-wise mapping).